### PR TITLE
style: Fix unnecessary-map (C417)

### PR DESCRIPTION
**Pull request by echoix**
Ruff rule: https://docs.astral.sh/ruff/rules/unnecessary-map/

Related Pylint rule: [consider-using-generator / R1728](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/consider-using-generator.html)

This is part of the effort to introduce Pylint 3.x for https://github.com/OSGeo/grass/issues/3921

### CODE REVIEW

File: gui/wxpython/gmodeler/model.py [hunk 1]
```diff
@@ -695,7 +695,7 @@ def Run(self, log, onDone, parent=None):
                     parent=parent,
                     message=_("Variables below not defined:")
                     + "\n\n"
-                    + "\n".join(map(lambda x: "%s: %s (%s)" % (x[0], x[1], x[2]), err)),
+                    + "\n".join(("%s: %s (%s)" % (x[0], x[1], x[2]) for x in err)),
```
Comment (ninsbl) [inline gui/wxpython/gmodeler/model.py:698]: Should not this be solved with an f-string?

File: gui/wxpython/gmodeler/model.py [hunk 2]
```diff
@@ -3499,21 +3495,21 @@ def cleanup():
                 r"""    %s("g.remove", flags="f", type="raster",
                 name=%s)
 """
-                % (run_command, ",".join(map(lambda x: '"' + x + '"', rast)))
+                % (run_command, ",".join(('"' + x + '"' for x in rast)))
```
Comment (ninsbl) [inline gui/wxpython/gmodeler/model.py:3498]: Also here, f-string seems more appropriate, no?

File: gui/wxpython/gmodeler/model.py [hunk 3]
```diff
@@ -3499,21 +3495,21 @@ def cleanup():
...
-                % (run_command, ",".join(map(lambda x: '"' + x + '"', vect)))
+                % (run_command, ",".join(('"' + x + '"' for x in vect)))
             )
         if rast3d:
             self.fd.write(
                 r"""    %s("g.remove", flags="f", type="raster_3d",
                 name=%s)
 """
-                % (run_command, ",".join(map(lambda x: '"' + x + '"', rast3d)))
+                % (run_command, ",".join(('"' + x + '"' for x in rast3d)))
```
Comment (ninsbl) [inline gui/wxpython/gmodeler/model.py:3512]: Again, f-string?

File: gui/wxpython/timeline/frame.py [hunk 4]
```diff
@@ -572,9 +572,7 @@ def SetDatasets(self, datasets):
             GError(parent=self, message=str(error), showTraceback=False)
             return
         self.datasets = datasets
-        self.datasetSelect.SetValue(
-            ",".join(map(lambda x: x[0] + "@" + x[1], datasets))
-        )
+        self.datasetSelect.SetValue(",".join((x[0] + "@" + x[1] for x in datasets)))
```
Comment (ninsbl) [inline gui/wxpython/timeline/frame.py:575]: F-string?

File: gui/wxpython/tplot/frame.py [hunk 5]
```diff
@@ -1268,15 +1268,15 @@ def SetDatasets(
             except:
                 self.coorval.SetValue(",".join(coors))
         if self.datasetsV:
-            vdatas = ",".join(map(lambda x: x[0] + "@" + x[1], self.datasetsV))
+            vdatas = ",".join((x[0] + "@" + x[1] for x in self.datasetsV))
```
Comment (ninsbl) [inline gui/wxpython/tplot/frame.py:1271]: F-string?

File: gui/wxpython/tplot/frame.py [hunk 6]
```diff
@@ -1268,15 +1268,15 @@ def SetDatasets(
...
             self.datasetSelectV.SetValue(vdatas)
             if attr:
                 self.attribute.SetValue(attr)
             if cats:
                 self.cats.SetValue(cats)
         if self.datasetsR:
             self.datasetSelectR.SetValue(
-                ",".join(map(lambda x: x[0] + "@" + x[1], self.datasetsR))
+                ",".join((x[0] + "@" + x[1] for x in self.datasetsR))
```
Comment (ninsbl) [inline gui/wxpython/tplot/frame.py:1279]: F-string?

**Review by ninsbl — COMMENTED** [review]
I guess some changes would be better served with f-strings...

File: gui/wxpython/tplot/frame.py (hunk 6 above)
Comment (echoix) [reply gui/wxpython/tplot/frame.py:1279]:
```suggestion
                ",".join(f"{x[0]}@{x[1]}" for x in self.datasetsR)
```

File: gui/wxpython/tplot/frame.py (hunk 5 above)
Comment (echoix) [reply gui/wxpython/tplot/frame.py:1271]:
```suggestion
            vdatas = ",".join(f"{x[0]}@{x[1]}" for x in self.datasetsV)
```

File: gui/wxpython/gmodeler/model.py (hunk 1 above)
Comment (echoix) [reply gui/wxpython/gmodeler/model.py:698]:
```suggestion
                    + "\n".join(f"{x[0]}: {x[1]} ({x[2]})" for x in err),
```

File: gui/wxpython/gmodeler/model.py (hunk 2 above)
Comment (echoix) [reply gui/wxpython/gmodeler/model.py:3498]:
```suggestion
                % (run_command, ",".join(f"\"{x}\"" for x in rast3d))
```

File: gui/wxpython/gmodeler/model.py (hunk 3 above)
Comment (echoix) [reply gui/wxpython/gmodeler/model.py:3512]:
```suggestion
                % (run_command, ",".join(f"\"{x}\"" for x in rast3d))
```

**Review by echoix — COMMENTED** [review]
@ninsbl I wasn't planning on solving some f-strings in here, as that's for a completely different PR, or maybe even PRs, as there are soooooooo many to fix, some in an automated way (ruff rule UP032, for example). But I still added some suggestions, even if I think that only doing one transformation at a time is easier to review.

**Review by ninsbl — APPROVED** [review]

**Pull request merged**